Convert product-elements/price to TypeScript

nielslange opened the pull request:
Fixes #7092

Currently, the `product-elements/price` is still written in JS and is using [prop-types](https://www.npmjs.com/package/prop-types) for type checking. This PR aims to convert `product-elements/price` to TypeScript.

### Testing

#### User Facing Testing

1. Create a test page and add the `Single Product block` to it.
2. Verify that the `Product Price inner block` is still visible.

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

nielslange committed:
Convert product-element/price to TypeScript

nielslange committed:
Apply feedback from #7095 to this PR

nielslange committed:
Merge branch 'trunk' into update/7092-convert-product-element-price-to-TS

nielslange committed:
Merge branch 'trunk' into update/7092-convert-product-element-price-to-TS

nielslange committed:
Export block due to Cross-Sells dependency

imanish003 commented on assets/js/atomic/blocks/product-elements/price/edit.tsx:
```diff
@@ -7,17 +7,38 @@ import {
+
+interface Props {
+	attributes: Attributes;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
```
Maybe we can use `BlockAttributes` from `./types` file as shown below. This will help with autocompletion. What do you think?

```suggestion
	setAttributes: ( attributes: Partial< BlockAttributes > & Record< string, unknown > ) => void;
```

nielslange replied:
Good catch, @imanish003. Let's definitely add `Partial< BlockAttributes >` here.

imanish003 reviewed (approved):
The changes look good to me. I just left a minor suggestion.

imanish003 commented on assets/js/atomic/blocks/product-elements/price/types.ts:
```diff
@@ -0,0 +1,5 @@
+export interface BlockAttributes {
```
I also noticed that `isDescendentOfQueryLoop` is missing from this interface which is present [here](https://github.com/woocommerce/woocommerce-blocks/blob/64684d4a9d6068aaf8189e5633691c9c08089d20/assets/js/atomic/blocks/product-elements/price/attributes.ts#L11).  May I know if there is any specific reason to omit it?

imanish003 replied:
@nielslange Tagging you here in case you missed this comment. 🙂

nielslange replied:
Thanks for your ping. Initially, I haven't added `isDescendentOfQueryLoop` as I thought it was a leftover of an older codebase because this attribute didn't appear in block.tsx. I just added the attribute accordingly.

nielslange committed:
Update assets/js/atomic/blocks/product-elements/price/edit.tsx

Co-authored-by: Manish Menaria <the.manish.menaria@gmail.com>

nielslange committed:
Merge branch 'trunk' into update/7092-convert-product-element-price-to-TS

github-actions[bot] committed:
bot: update checkstyle.xml

nielslange committed:
Apply review feedback

nielslange committed:
Outsource supports section

nielslange committed:
Merge branch 'update/7092-convert-product-element-price-to-TS' of github.com:woocommerce/woocommerce-blocks into update/7092-convert-product-element-price-to-TS

# Conflicts:
#	assets/js/atomic/blocks/product-elements/price/edit.tsx

github-actions[bot] committed:
bot: update checkstyle.xml

nielslange commented:
> The changes look good to me. I just left a minor suggestion.

Thanks for your review, @imanish003. I addressed all your feedback. Would you mind doing another review?

nielslange committed:
Merge branch 'trunk' into update/7092-convert-product-element-price-to-TS
